RoundsWithFriends/Patches/MapManager.cs

```diff
@@ -0,0 +1,276 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Collections;
+using Sonigon;
+using HarmonyLib;
+using UnboundLib;
+using UnboundLib.Networking;
+using System.Reflection;
+using UnityEngine;
+using Photon.Pun;
+
+namespace RWF.Patches
+{
+    [HarmonyPatch(typeof(MapManager), "CallInNewMapAndMovePlayers")]
+    class MapManager_Patch_CallInNewMapAndMovePlayers
+    {
+        static void Prefix(MapManager __instance, PhotonView ___view, int mapID)
+        {
+            if (!PhotonNetwork.IsMasterClient && !PhotonNetwork.OfflineMode)
+            {
+                return;
+            }
+            NetworkingManager.RPC(typeof(GeneralizedSpawnPositions), nameof(GeneralizedSpawnPositions.RPCA_SetSeed), new object[] { UnityEngine.Random.Range(0, 10000) });
+        }
+    }
+    class GeneralizedSpawnPositions
+    {
+        private static int seed = 0;
+        internal static void SetSeed(int newSeed)
+        {
+            seed = newSeed;
+        }
+        [UnboundRPC]
+        internal static void RPCA_SetSeed(int seed)
+        {
+            GeneralizedSpawnPositions.SetSeed(seed);
+        }
+
+        static int NumberOfTeams => TeamIDs.Count();
+        static int[] TeamIDs => PlayerManager.instance.players.Select(p => p.teamID).Distinct().ToArray();
+
+        private const float range = 2f;
+        private const float maxProject = 1000f;
+        private const float groundOffset = 1f;
+        private const float maxDistanceAway = 5f;
+        private const int maxAttempts = 1000;
+        private const int numSamples = 100;
+        private const float eps = 0.1f;
+        private const float lmargin = 0.1f;
+        private const float rmargin = 0.1f;
+        private const float tmargin = 0.2f;
+        private const float bmargin = 0f;
+        private const float minDistanceFromLedge = 1f;
+        private static LayerMask groundMask = (LayerMask) LayerMask.GetMask(new string[] { "Default", "IgnorePlayer" });
+        //private const float minDistance = 5f;
+        internal static Dictionary<Player, Vector2> GetSpawnDictionary(List<Player> players, SpawnPoint[] spawnPoints)
+        {
+            // initialize the RNG to sync clients
+            UnityEngine.Random.InitState(seed);
```

Comment:
We should probably use an instance of `System.Random` here. Setting the global `UnityEngine.Random` seed to one of just 10000 possibilities probably screws up a lot of things in a subtle way. For example card choices are random and also based on `UnityEngine.Random`, so setting the seed basically locks card choices to one of 10000 paths 😄